cmd/duffle/pull.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/deislabs/duffle/pkg/duffle/home"
+	"github.com/deislabs/duffle/pkg/reference"
+
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/cnab-to-oci/remotes"
+	"github.com/spf13/cobra"
+)
+
+type pullCmd struct {
+	output             string
+	targetRef          string
+	insecureRegistries []string
+	home               home.Home
+}
+
+func newPullCmd(w io.Writer) *cobra.Command {
+	const usage = `Pulls a bundle from an OCI repository`
+	const pullDesc = `
+Pulls a CNAB bundle from an OCI repository.
+The only argument for this command is the repository where
+the bundle can be found, and by default, this command pulls the
+bundle and stores it in the local bundle store.
```

Comment:
Does this command do anything with images referenced by the bundle?

Reply:
No, by default it does _not_ pull the referenced images.

Reply:
I’m revisiting this PR after the whole OCI discussion - do you think it pulling the bundle should pull the referenced images?
Most likely they will not be used on the same machine that pulls the bundle, so not sure about this - would an optional flag for pulling the images be helpful?

Reply:
I think it makes sense for a local development use case to pre-populate the local docker image store (just pull the bundle and everything in it and it "works"), so I agree with the optional flag.

By the way I don't think it should be a `cnab-to-oci` feature, as it has no knowledge of the docker engine, only duffle has. I guess it will then parse the `bundle.json` and call `docker pull` on every docker image using the docker client ?

Reply:
Yeah, I agree - I wasn't thinking this should be in `cnab-to-oci`.